editor/editor_file_system.cpp

```diff
@@ -1816,13 +1822,19 @@ void EditorFileSystem::update_files(const Vector<String> &p_script_paths) {
 			if (fs->files[cpos]->type == SNAME("PackedScene")) {
 				_queue_update_scene_groups(file);
 			}
+			if (fs->files[cpos]->type == SNAME("Resource") || old_script_class_icon_path != fs->files[cpos]->script_class_icon_path) {
+				clear_icon_cache = true;
+			}
 			updated = true;
 		}
 	}
 
 	if (updated) {
 		_update_pending_script_classes();
 		_update_pending_scene_groups();
+		if (clear_icon_cache) {
+			FileSystemDock::get_singleton()->clear_icon_cache();
```

Comment:
Instead of calling it here, could it be a callback for `filesystem_changed` instead?

Reply:
I'm trying to clear the icon cache in FileSystemDock only if resource are changed. If I clear the cache on every `filesystem_changed` it will be the same as before where the cache was cleared at each tree update because `filesystem_changed` is signaled everytime something is saved. I could created a new signal just for icon_changed if it's better?

Reply:
tbh it would be cleaner if EditorFileSystem was responsible only for file system. So yeah, it's better to have a new signal.

Reply:
You are right, even a PackedScene is a Resource. I will leave it like that.

Reply:
I took your alternative solution and moved the icon_path directly into the EditorFileSystem.